src/CyBy/Draw/MoleculeCanvas.idr

```diff
@@ -20,6 +20,20 @@ import Text.SVG
 %language ElabReflection
 %hide Language.Reflection.TTImp.Mode
 
+
+-- Record pairing graph with one of its node's index
+record DepNode where
+  constructor DN
+  graph : CDGraph
+  node  : Fin graph.order
+
+%runElab derive "DepNode" [Show]
+
+-- TODO: Is this correct?? The order has to be the same because `g1 == g2`
```

Comment:
This is correct. The whole point is that from `g1 == g2` does *not* follow `g1.order === g2.order` (the `===` is not a typo here). At least not for Idris via propositional equality. That's the reason why we can't derive `Eq` automatically. Since this is correct, please remove this comment.